Engine_Revit_UI/Modify/SetParameter.cs

```diff
@@ -133,9 +133,40 @@ public static bool SetParameter(this Element element, BuiltInParameter builtInPa
 
         public static bool SetParameter(this Parameter parameter, object value, Document document = null)
         {
+            // Skip non-workshared documents because they can't contain worksets
+            if (document.IsWorkshared == false)
+                return false;
+
+            // Skip non-string workset names
+            string worksetName = value as string;
+            if (worksetName is string == false)
+                return false;
+
+            // Skip null or empty workset names
+            if (string.IsNullOrWhiteSpace(worksetName))
+                return false;
+
+            // Skip null and read-only parameters
             if (parameter == null || parameter.IsReadOnly)
                 return false;
 
+            // Workset parameters
+            // Filter for worksets
+            if (parameter.Id.IntegerValue == (int)BuiltInParameter.ELEM_PARTITION_PARAM)
+            {
+                FilteredWorksetCollector worksets = new FilteredWorksetCollector(document).OfKind(WorksetKind.UserWorkset); // Find all user worksets
+
+                Workset workset = worksets.Where(x => x.Name == worksetName).FirstOrDefault(); // Find the specified workset if it exists
```

Comment:
Here I would call `Create.Workset(document, worksetName)` method that would either return an existing workset with given name or a newly created one. Next step would be to set this workset to parameter.

Reply:
Revised accordingly